### PR TITLE
Automated releases for api-gateway component

### PR DESCRIPTION
**Pull request by werdes72**
See: https://github.com/kyma-project/api-gateway/issues/7

### CODE REVIEW

File: templates/data/generic_component_api_gateway_data.yaml [hunk 1]
```diff
@@ -30,8 +35,29 @@ templates:
...
                     - disable_testgrid
-                    - image_buildpack-golang-kubebuilder2
+                    - image_buildpack-golang-toolbox
+              - jobConfig:
+                  name: rel-api-gateway
+                  args:
+                    - "/home/prow/go/src/github.com/kyma-project/api-gateway"
+                    - "ci-release"
+                  branches:
+                    - "^\\d+\\.\\d+\\.\\d+(?:-.*)?$"
```
Comment (Ressetkk) [inline templates/data/generic_component_api_gateway_data.yaml:49]:
```suggestion
                    - `^\d+\.\d+\.\d+(?:-.*)?$`
```

File: templates/data/generic_component_api_gateway_data.yaml [hunk 2]
```diff
@@ -2,6 +2,11 @@ templates:
   - from: generic.tmpl
     render:
       - to: ../../prow/jobs/api-gateway/api-gateway-build.yaml
+        localSets:
+          job_spec:
+            command: "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/build-generic.sh"
+            env:
+              GO111MODULE: "on"
```
Comment (Ressetkk) [inline templates/data/generic_component_api_gateway_data.yaml:9]:
```suggestion

```

Comment (werdes72) [reply templates/data/generic_component_api_gateway_data.yaml:9]: `2022/09/06 10:37:40 Cannot parse data file templates/data/generic_component_api_gateway_data.yaml: yaml: line 47: found character that cannot start any token`

File: templates/data/generic_component_api_gateway_data.yaml (hunk 1 above)
Comment (werdes72) [reply templates/data/generic_component_api_gateway_data.yaml:49]: `2022/09/06 10:37:40 Cannot parse data file templates/data/generic_component_api_gateway_data.yaml: yaml: line 47: found character that cannot start any token`

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**